pil_analyzer/src/pil_analyzer.rs

```diff
@@ -188,6 +189,34 @@ impl<T: FieldElement> PILContext<T> {
             PilStatement::ConstantDefinition(_, name, value) => {
                 self.handle_constant_definition(name, value)
             }
+            PilStatement::LetStatement(start, name, None) => {
+                // Handle all let statements without assignment as witness column declarations for now.
+                self.handle_polynomial_declarations(
+                    self.to_source_ref(start),
+                    vec![PolynomialName {
+                        name,
+                        array_size: None,
+                    }],
+                    PolynomialType::Committed,
+                )
+            }
+            PilStatement::LetStatement(start, name, Some(value)) => {
```

Comment:
How would we support intermediate polynomials here? It seems like all cases lead to fixed? It seems like this depends on whether the rhs is constant or not?

Reply:
Yes, we need a separate analysis pass for that. We also need a more complicated analysis pass to distinguish witness columns from fixed columns, currently, it's only syntax. For example if we have
```
let f;
f = g;
let g = |i| i + 2;
```
then both g and f should be fixed

Reply:
Right. I have PRs adding intermediate polynomials, it seems like with this they would be turned into constants then? How do we avoid that?

Reply:
We do a proper analysis.

Reply:
And what we'd do is:
- If a polynomial has no definition, it is a witness polynomial.
- If a polynomial is defined in terms of a lambda function, it is constant.
- If it is defined in terms of other polynomials, then:
  - If all of them are fixed, should be fixed as well.
  - If any of them is witness or intermediate, it is intermediate.

Is that how it would work?